### ISSUE TITLE
Return list of expanded URLs instead of just the string of only the first URL

### ISSUE THREAD

**Issue by hauselin** [HumanCompatibleAI/ranking-challenge#14]
Hi, I noticed in the line below the code only returns the first expanded URL (`urls[0]`), and returns it as a string. Often, there are multiple URLs, so it would be great if all expanded URLs in any given tweet were returned. See the suggestion below. Thanks! 

https://github.com/HumanCompatibleAI/ranking-challenge/blob/28b900e18b7aa44855309b2bbb164a2365ffc75d/sample_data/preprocessing.py#L186

```python
# current: returns string and returns only the first expanded URL
expanded_url = urls[0].get("expanded_url", None) if urls else None

# suggestion: returns list of all expanded URLs
expanded_urls = [url.get("expanded_url") for url in urls if url.get("expanded_url")]
```

**Comment by raindrift** [comment]
Agreed, this is the right thing. In the current version of the api spec, the urls are a list. The sample data generator doesn't insert multiple urls yet though, so I'll leave this issue open until it does.

**Comment by hauselin** [comment]
@raindrift, I also noticed two additional/related issues in `integration_test.py` (and maybe also other places in the code base?). Right now, when running tests on Twitter, `embedded_urls` contain only shortened URLs and misses all the URLs in  `expanded_url` in the raw data. 

1. The current `re.findall` solution will miss URLs that don't start with `"https"` or `"http"`. I believe sometimes, on certain platforms, URLs might omit the `http`?

```python
text = "https://www.facebook.com www.google.com github.com"  # 3 URLS 

# current
>>> re.findall(r"https?://(?:[-\w.]|(?:%[\da-fA-F]{2}))+", text)
['https://www.facebook.com']  # misses 2 of the 3 URLs

# more robust solution: https://urlextract.readthedocs.io/en/latest/urlextract.html
from urlextract import URLExtract
extractor = URLExtract()
>>> extractor.find_urls(text)
['https://www.facebook.com', 'www.google.com', 'github.com']  # gets all 3 URLS
```

2. For Twitter, only shortened URLs are returned; the expanded URLs in `expanded_url` field isn't returned/included. For example, it seems like the current implementation of `data_pull.py` will always return `None` (`row.get("expanded_url", None)`) because `expanded_url` is inside `entities` and in a list of dictionaries. Have to extract/unpack them properly (see example code). 

https://github.com/HumanCompatibleAI/ranking-challenge/blob/ae88eb09d611b1175c9668cb6a7b2cf5c4a2f419/sample_data/data_pull.py#L249-L253

```python
embedded_urls = []
urls = row.entities.get("urls", [])
for url in urls:
    expanded_url = url.get("expanded_url")
    if expanded_url:
        embedded_urls.append(expanded_url)
```
